test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs

```diff
@@ -482,6 +483,31 @@ public void OnProvidersExecuting_PreservesBindingInfo_WhenInferringFor_Parameter
             Assert.Equal("top", bindingInfo.BinderModelName);
         }
 
+        [Fact]
+        public void OnProvidersExecuting_PreservesBindingInfo_WhenInferringFor_ParameterWithModelBinder_WithoutExplicitName_OnCollection()
```

Comment:
This is not the addition I was looking for in my earlier [comment](https://github.com/aspnet/Mvc/pull/7967/files#r197895856). Need to exercise your changes to `InferBoundPropertyModelPrefixes(...)`. That is, model the new test after `InferBoundPropertyModelPrefixes_SetsModelPrefix_ForComplexTypeFromValueProvider()` but use a property of a collection type (array, list, whatever).